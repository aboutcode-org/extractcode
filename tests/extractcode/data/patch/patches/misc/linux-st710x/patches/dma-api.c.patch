--- linux-orig/arch/sh/drivers/dma/dma-api.c	2007-12-06 10:57:14.000000000 +0100
+++ linux/arch/sh/drivers/dma/dma-api.c	2007-12-06 11:21:31.000000000 +0100
@@ -388,8 +388,10 @@
 			chan->flags |= DMA_TEI_CAPABLE;
 
 		init_waitqueue_head(&chan->wait_queue);
+#ifdef CONFIG_SYSFS
 		dma_create_sysfs_files(chan, info);
-	}
+#endif
+        }
 
 	list_add(&info->list, &registered_dmac_list);
 
@@ -399,12 +401,14 @@
 
 void unregister_dmac(struct dma_info *info)
 {
+#ifdef CONFIG_SYSFS
 	unsigned int i;
 
 	for (i = 0; i < info->nr_channels; i++)
 		dma_remove_sysfs_files(info->channels + i, info);
+#endif
 
-	if (!(info->flags & DMAC_CHANNELS_CONFIGURED))
+        if (!(info->flags & DMAC_CHANNELS_CONFIGURED))
 		kfree(info->channels);
 
 	list_del(&info->list);
