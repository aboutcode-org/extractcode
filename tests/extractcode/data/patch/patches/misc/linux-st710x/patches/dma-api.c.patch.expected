[('linux-orig/arch/sh/drivers/dma/dma-api.c',
  'linux/arch/sh/drivers/dma/dma-api.c',
  '--- linux-orig/arch/sh/drivers/dma/dma-api.c\n+++ linux/arch/sh/drivers/dma/dma-api.c\n@@ -388,8 +388,10 @@\nchan->flags |= DMA_TEI_CAPABLE;\n\ninit_waitqueue_head(&chan->wait_queue);\n+#ifdef CONFIG_SYSFS\ndma_create_sysfs_files(chan, info);\n-\t}\n+#endif\n+        }\n\nlist_add(&info->list, &registered_dmac_list);\n\n@@ -399,12 +401,14 @@\n\nvoid unregister_dmac(struct dma_info *info)\n{\n+#ifdef CONFIG_SYSFS\nunsigned int i;\n\nfor (i = 0; i < info->nr_channels; i++)\ndma_remove_sysfs_files(info->channels + i, info);\n+#endif\n\n-\tif (!(info->flags & DMAC_CHANNELS_CONFIGURED))\n+        if (!(info->flags & DMAC_CHANNELS_CONFIGURED))\nkfree(info->channels);\n\nlist_del(&info->list);')]